gnovm/stdlibs/std/native.go

```diff
@@ -32,6 +32,20 @@ func GetHeight(m *gno.Machine) int64 {
 	return m.Context.(ExecContext).Height
 }
 
+func getTimestamp(m *gno.Machine) int64 {
+	return m.Context.(ExecContext).Timestamp
+}
+
+// getPrevFunctionNameFromTarget returns the last called function name (identifier) from the call stack.
+func getPrevFunctionNameFromTarget(m *gno.Machine, targetFunc string) string {
+	for i := 0; i < len(m.Frames); i++ {
+		if m.Frames[i].Func.Name == gno.Name(targetFunc) && i > 0 {
```

Comment:
Maybe In some cases, m.Frames[i].Func could be nil. Should we check for it to prevent a panic if it is nil?

Reply:
I don't think that a frame without a function is possible. I'd rather not include a check and panic unless a valid use case for a functionless frame is identified. Any thoughts on this?
cc @gnolang/core-contributors

Reply:
I notice that we are pushing some Frame with Func equal to nil: https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/machine.go#L1722-L1737 https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/machine.go#L1665-L1674 
However, I am unsure if it is typical in this machine state to have these two types of frames.

Reply:
I also think it is unlikely happened. The only possible case would be if `std.Emit` is called independently in the global scope, not within any function.
```go
package foo

import "std"

std.Emit(/* ... */)
^^^^^^^^
    syntax error: non-declaration statement outside function body

```

However, non-declaration statement cannot be allowed, an error will occur in this case. So, I guess it seems to be no need to specifically nil check for here.

Reply:
```go
# start a new node
gnoland start

gnokey maketx addpkg -pkgdir $WORK -pkgpath gno.land/r/demo/cbee -gas-fee 1000000ugnot -gas-wanted 2500000 -broadcast -chainid=tendermint_test test1
stdout OK!
stdout 'GAS WANTED: 2500000'
stdout 'GAS USED: [0-9]+'
stdout 'EVENTS: \[\]'

gnokey maketx call -pkgpath gno.land/r/demo/cbee -func Foo -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout OK!
stdout 'GAS WANTED: 2000000'
stdout 'GAS USED: [0-9]+'
stdout 'EVENTS: \[?.*\]'

-- cbee.gno --
package cbee

import (
    "std"
)

const (
	sender = "sender"
	receiver = "receiver"
)

type contractA struct{}

func (c *contractA) sender(cb func()) {
	subSender()
	cb()
}

func subSender() {
	std.Emit(sender, "k1", "v1", "k2", "v2")
}

type contractB struct{}

func (c *contractB) subReceiver() {
	std.Emit(receiver, "bar", "baz")
}

func Foo() {
    a := &contractA{}
	b := &contractB{}
	for i := 0; i < 2; i++ {
		a.sender(func() {
			b.subReceiver()
		})
	}

}
```
Maybe in this case we can have the issue ?

Reply:
This is an issue that needs to be resolved.  See my earlier comment https://github.com/gnolang/gno/pull/1653#discussion_r1572680102

Reply:
Yes this were definitely have a case where it crashed. Now I've fix this to include a nil check. thanks @omarsy